File: core/src/main/java/io/outright/xj/core/model/Audio.java
// Copyright Outright Mental, Inc. All Rights Reserved.
package io.outright.xj.core.model;

import io.outright.xj.core.model.chord.Chord;
import io.outright.xj.core.model.event.Event;

public interface Audio {
  /**
   * Audio Waveform (path in filesystem)
   *
   * @return
   * TODO replace with Waveform (String, filesystem) type
   */
  String Waveform();

  /**
   * Audio Length (seconds)
   *
   * @return
   * TODO replace with Length (seconds) type
   */
  float Length();

  /**
   * Audio Start (seconds from beginning of waveform)
   *
   * @return
   * TODO replace with Time (float, seconds) type
   */
  float Start();

  /**
   * Audio Tempo (beats per minute)
   *
   * @return
   * TODO replace with Tempo (float) type
   */
  float Tempo();

  /**
   * Audio Pitch (Hz)
   *
   * @return float
   * TODO replace with Pitch (float, Hz) type
   */
  float Pitch();

  /**
   * Audio Chords
   *
   * @return Chord[]
   */
  Chord[] Chords();

  /**
   * Audio Events
   *
   * @return Event[]
   */
  Event[] Events();
}

File: ui/app/models/audio.js
// Copyright (c) 2017, Outright Mental Inc. (http://outright.io) All Rights Reserved.
import DS from 'ember-data';

export default DS.Model.extend({
  instrument: DS.belongsTo({}),
  name: DS.attr('string'),
  waveformUrl: DS.attr('string'),
  start: DS.attr('number'),
  length: DS.attr('number'),
  tempo: DS.attr('number'),
  pitch: DS.attr('number'),
});
